compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6897,11 +6896,16 @@ private BType checkInvocationParam(BLangInvocation iExpr, AnalyzerData data) {
                             incRecordAllowAdditionalFields = false;
                         }
                         iExpr.requiredArgs.add(expr);
+                    } else if (invokableSymbol.restParam != null &&
+                            invokableSymbol.restParam.name.value.equals(((BLangNamedArgsExpression) expr).name.value)) {
+                        // can not provide a rest parameters as named args
+                        checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
+                        dlog.error(expr.pos, DiagnosticErrorCode.REST_ARG_CANNOT_BE_NAMED_ARG, iExpr.name.value);
```

Comment:
this is same as the else block logic. only different is the `DiagnosticErrorCode`. We can merge two blocks.

Reply:
Since there is an extra `if` inside else block can we merge these without doing same check twice? Can maybe assign the condition to a var and use it